src/components/application_manager/test/commands/command_impl_test.cc

```diff
@@ -0,0 +1,279 @@
+  EXPECT_TRUE(command->Init());
+  EXPECT_TRUE(command->CheckPermissions());
+  EXPECT_TRUE(command->CleanUp());
+
+  // Default value of `allowed_to_terminate_` is true
+  EXPECT_TRUE(command->AllowedToTerminate());
+  command->SetAllowedToTerminate(false);
+  EXPECT_FALSE(command->AllowedToTerminate());
+
+  const uint32_t kCorrelationId = 3u;
+  const int32_t kFunctionId = 4u;
+  const uint32_t kConnectionKey = 5u;
+
+  (*msg)[strings::params][strings::correlation_id] = kCorrelationId;
+  (*msg)[strings::params][strings::function_id] = kFunctionId;
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey;
+
+  EXPECT_EQ(kDefaultTimeout_, command->default_timeout());
+  EXPECT_EQ(kCorrelationId, command->correlation_id());
+  EXPECT_EQ(kConnectionKey, command->connection_key());
+  EXPECT_EQ(kFunctionId, command->function_id());
+  EXPECT_NO_THROW(command->Run());
+  EXPECT_NO_THROW(command->onTimeOut());
+}
+
+TEST_F(CommandImplTest, ReplaceMobileByHMIAppId_NoAppIdInMessage_UNSUCCESS) {
+  MessageSharedPtr msg;
+  UCommandImplPtr command = CreateCommand<UCommandImpl>(msg);
+
+  EXPECT_CALL(app_mngr_, application(_)).Times(0);
+
+  command->ReplaceMobileByHMIAppId(*msg);
+}
+
+TEST_F(CommandImplTest, ReplaceMobileByHMIAppId_SUCCESS) {
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::app_id] = kAppId1_;
+
+  UCommandImplPtr command = CreateCommand<UCommandImpl>(msg);
+
+  MockAppPtr app = CreateMockApp();
+
+  EXPECT_CALL(app_mngr_, application(kAppId1_)).WillOnce(Return(app));
+  ON_CALL(*app, hmi_app_id()).WillByDefault(Return(kAppId2_));
+
+  command->ReplaceMobileByHMIAppId(*msg);
+
+  EXPECT_EQ(kAppId2_, (*msg)[strings::app_id].asUInt());
+}
+
+TEST_F(CommandImplTest, ReplaceMobileByHMIAppId_Array_SUCCESS) {
+  MessageSharedPtr msg = CreateArrayMessage(kDefaultMsgCount_);
+  UCommandImplPtr command = CreateCommand<UCommandImpl>(msg);
+
+  MockAppPtr app = CreateMockApp();
+
+  EXPECT_CALL(app_mngr_, application(_))
+      .Times(kDefaultMsgCount_)
+      .WillRepeatedly(Return(app));
+  ON_CALL(*app, hmi_app_id()).WillByDefault(Return(kAppId2_));
+
+  command->ReplaceMobileByHMIAppId(*msg);
+
+  EXPECT_TRUE(msg->asArray());
+  smart_objects::SmartArray::const_iterator it(msg->asArray()->begin());
+  const smart_objects::SmartArray::const_iterator it_end(msg->asArray()->end());
+  for (; it != it_end; ++it) {
+    EXPECT_EQ(kAppId2_, (*it)[strings::app_id].asUInt());
+  }
+}
+
+TEST_F(CommandImplTest, ReplaceMobileByHMIAppId_Map_SUCCESS) {
+  MessageSharedPtr msg = CreateMapMessage(kDefaultMsgCount_);
+  UCommandImplPtr command = CreateCommand<UCommandImpl>(msg);
+
+  MockAppPtr app = CreateMockApp();
+
+  EXPECT_CALL(app_mngr_, application(_))
+      .Times(kDefaultMsgCount_)
+      .WillRepeatedly(Return(app));
+  ON_CALL(*app, hmi_app_id()).WillByDefault(Return(kAppId2_));
+
+  command->ReplaceMobileByHMIAppId(*msg);
+
+  std::set<std::string> keys(msg->enumerate());
```

Comment:
@LuxoftAKutsan, `set` been used because `enumerate()` returns `set` of map keys stored inside `std::string`.